packages/manager/src/components/ImageSelect/ImageSelect.tsx

```diff
@@ -85,10 +87,22 @@ export const imagesToGroupedItems = (images: Image[]) => {
         draft.push({
           label: thisGroup,
           options: group
+            .filter((thisImage) => {
+              const { eol } = thisImage;
+              const diff = DateTime.now().diff(DateTime.fromISO(eol!), 'months')
+                .months;
+              return diff > MAX_MONTHS_EOL_FILTER && eol !== null
+                ? false
+                : true;
```

Comment:
The non-null assertion (`!`) caught my attention. I found that when `eol` was `null`, `diff` would be `NaN`. Optionally, we could check if it is null before we calculate the time difference. This change is up to you, it results in the same behavior.



```suggestion
              const { eol } = thisImage;
              if (!eol) {
                // no end of life is known so show Image to be safe
                return true;
              }
              const diff = DateTime.now().diff(DateTime.fromISO(eol), 'months')
                .months;

              // if image is past its end of life, hide it, otherwise show it
              return diff > MAX_MONTHS_EOL_FILTER ? false : true;
```